File: src/examples/USZipCodeSingleLookupExample.cs
namespace Examples
{
	using System;
	using System.IO;
    using System.Net;
    using SmartyStreets;
	using SmartyStreets.USZipCodeApi;

	internal static class USZipCodeSingleLookupExample
	{
		public static void Run()
		{
			// You don't have to store your keys in environment variables, but we recommend it.
			var authId = Environment.GetEnvironmentVariable("SMARTY_AUTH_ID");
			var authToken = Environment.GetEnvironmentVariable("SMARTY_AUTH_TOKEN");

			using var client = new ClientBuilder(authId, authToken).BuildUsZipCodeApiClient();

			// Documentation for input fields can be found at:
			// https://smartystreets.com/docs/us-zipcode-api#input-fields

			var lookup = new Lookup
			{
				InputId = "dfc33cb6-829e-4fea-aa1b-b6d6580f0817", // Optional ID from your system
				City = "Mountain View",
				State = "California",
				ZipCode = "94039"
			};

			//uncomment the line below to add a custom parameter
			//lookup.AddCustomParameter("zipcode", "94039");
			
			var batch = new Batch();
			batch.Add(lookup);
			
			try
			{
				client.Send(batch);
			}
			catch (SmartyException ex)
			{
				Console.WriteLine(ex.Message);
				Console.WriteLine(ex.StackTrace);
				return;
			}
			catch (IOException ex)
			{
				Console.WriteLine(ex.StackTrace);
				return;
			}

			var result = lookup.Result;
			var cities = result.CityStates;
			var zipCodes = result.ZipCodes;

			if (cities == null || zipCodes == null)
			{
				Console.WriteLine("No results.");
				return;
			}

			Console.WriteLine("Input ID: " + result.InputId);

			foreach (var city in cities)
			{
				Console.WriteLine("\nCity: " + city.City);
				Console.WriteLine("State: " + city.State);
				Console.WriteLine("Mailable City: " + city.MailableCity);
			}

			foreach (var zipCode in zipCodes)
			{
				Console.WriteLine("\nZIP Code: " + zipCode.ZipCode);
				Console.WriteLine("Latitude: " + zipCode.Latitude);
				Console.WriteLine("Longitude: " + zipCode.Longitude);
			}
		}
	}
}

File: src/vbexamples/USZipCodeSingleLookupExample.vb
Imports System.IO
Imports SmartyStreets
Imports SmartyStreets.USZipCodeApi

Module USZipCodeSingleLookupExample

    Dim authID = Environment.GetEnvironmentVariable("SMARTY_AUTH_ID")
    Dim authToken = Environment.GetEnvironmentVariable("SMARTY_AUTH_TOKEN")

    Dim client = New ClientBuilder(authID, authToken).WithLicense(New List(Of String) From {"us-core-cloud"}).BuildUsZipCodeApiClient()

    Sub USZipCodeSingleLookupExample()

        Dim lookup As New Lookup()
        With lookup
            .InputId = "dfc33cb6-829e-4fea-aa1b-b6d6580f0817"
            .City = "Mountain View"
            .State = "California"
            .ZipCode = "94039"
        End With

        Console.WriteLine("*******************************************************")
        Console.WriteLine()

        Try
            client.Send(lookup)
        Catch ex As SmartyException
            Console.WriteLine(ex.Message)
            Console.WriteLine(ex.StackTrace)
        Catch ex As IOException
            Console.WriteLine(ex.StackTrace)
        Catch ex As Exception
            Console.WriteLine(ex.Message)
            Console.WriteLine(ex.StackTrace)
        End Try

        Dim result = lookup.Result
        Dim cities = result.CityStates
        Dim zipCodes = result.ZipCodes

        If result.Status IsNot Nothing Then
            Console.WriteLine("Lookup has an invalid status.")
            Console.WriteLine("Status: " + result.Status)
            Console.WriteLine("Reason: " + result.Reason + Environment.NewLine)
            Return
        End If

        If zipCodes Is Nothing Or cities Is Nothing Then
            Console.WriteLine("Lookup has no candidates. The lookup is not valid." + Environment.NewLine)
            Return
        End If

        Console.WriteLine("Lookup is valid." + Environment.NewLine())

        Console.WriteLine("Input ID: " + result.InputId)

        Console.WriteLine(CStr(cities.Length) + " city and state match" + If(cities.Length = 1, ":", "es:"))

        For Each city In cities
            Console.WriteLine()
            Console.WriteLine("City: " + city.City)
            Console.WriteLine("State: " + city.State)
            Console.WriteLine("Mailable City: " + CStr(city.MailableCity))
        Next

        For Each zipCode In zipCodes
            Console.WriteLine()
            Console.WriteLine("ZIP Code: " + zipCode.ZipCode)
            Console.WriteLine("Latitude: " + CStr(zipCode.Latitude))
            Console.WriteLine("Longitude: " + CStr(zipCode.Longitude))
        Next

        Console.WriteLine()

    End Sub

End Module
